src/supautils.c

```diff
@@ -368,15 +374,19 @@ look_for_reserved_role(Node *utility_stmt, List *roles_list)
 					RoleSpec *role = lfirst(item);
 					ListCell *role_cell;
 
-					// Break if the role is PUBLIC, let pg give a better error later
-					if (role->roletype == ROLESPEC_PUBLIC)
+					// Break if the role is PUBLIC, CURRENT_USER or SESSION_USER
+					// these cannot be used anyway because doing DROP ROLE CURRENT_USER
+					// makes pg fail with
+					// ERROR:  cannot use special role specifier in DROP ROLE
+					if (role->roletype != ROLESPEC_CSTRING)
 						break;
 
 					foreach(role_cell, roles_list)
 					{
 						char *reserved_role = (char *) lfirst(role_cell);
-
-						if (strcmp(get_rolespec_name(role), reserved_role) == 0)
+						// We don't use get_rolespec_name() here because it'll
+						// validate the role exists and will make DROP ROLE IF EXISTS fail
```

Comment:
Similarly, this comment can be improved as:

```
/*
 * We extract the string from RoleSpec, instead of using get_rolespec_name(),
 * because that function does a syscache lookup, and that would fail for
 * non-existent roles. E.g when IF EXISTS clause is being used, which is not
 * supposed to ERROR on non-existent roles. We leave handling of
 * non-existent roles up to Postgres.
 */
```

Reply:
These are great comments :100: , I've just added them. Thanks Gurjeet!